staging/src/k8s.io/legacy-cloud-providers/openstack/metadata.go

```diff
@@ -67,6 +84,13 @@ type DeviceMetadata struct {
 	// .. and other fields.
 }
 
+type ec2Metadata struct {
+	PublicIP     string `json:"public-ipv4"`
+	LocalIP      string `json:"local-ipv4"`
+	InstanceType string `json:"instance-type"`
+	// .. and other fields we don't care about.  Expand as necessary.
+}
+
 // Metadata has the information fetched from OpenStack metadata service or
 // config drives. Assumes the "2012-08-10" meta_data.json format.
```

Comment:
I think this "assumes 2012-08-10 format" comment is incorrect.  Remove (or update).